src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java

```diff
@@ -37,12 +38,30 @@ public void ifAcquiredOrElse(
         Consumer<String> onSuccess,
         Consumer<BlobErrorCode> onFailure,
         boolean releaseLease
+    ) {
+        ifAcquiredOrElse(
+            blobClient,
+            blob -> true,
+            onSuccess,
+            onFailure,
+            releaseLease
+        );
+    }
+
+    public void ifAcquiredOrElse(
+        BlobClient blobClient,
+        Predicate<BlobClient> blobCondition,
```

Comment:
I disagree with this change - it's unnecessary and adds more complexity to the code that should only deal with acquiring leases. Why not just have this condition in the actual business logic?